homeassistant/components/overkiz/config_flow.py

```diff
@@ -0,0 +1,110 @@
+
+from .const import CONF_HUB, DEFAULT_HUB, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Overkiz (by Somfy)."""
+
+    VERSION = 1
+
+    async def async_validate_input(self, user_input: dict[str, Any]) -> None:
+        """Validate user credentials."""
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        server = SUPPORTED_SERVERS[user_input[CONF_HUB]]
+
+        async with OverkizClient(
+            username=username, password=password, server=server
+        ) as client:
+            await client.login()
+
+            # Set first gateway as unique id
+            if gateways := await client.get_gateways():
+                gateway_id = gateways[0].id
+                await self.async_set_unique_id(gateway_id)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step via config flow."""
+        errors = {}
+
+        if user_input:
+            try:
+                await self.async_validate_input(user_input)
+            except TooManyRequestsException:
+                errors["base"] = "too_many_requests"
+            except BadCredentialsException:
+                errors["base"] = "invalid_auth"
+            except (TimeoutError, ClientError):
+                errors["base"] = "cannot_connect"
+            except MaintenanceException:
+                errors["base"] = "server_in_maintenance"
+            except Exception as exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+                _LOGGER.exception(exception)
+            else:
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=user_input[CONF_USERNAME], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Required(CONF_HUB, default=DEFAULT_HUB): vol.In(
+                        {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
+                    ),
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP discovery."""
+        hostname = discovery_info.hostname
+        gateway_id = hostname[8:22]
+
+        _LOGGER.debug("DHCP discovery detected gateway %s", obfuscate_id(gateway_id))
+
+        if self._gateway_already_configured(gateway_id):
+            _LOGGER.debug("Gateway %s is already configured", obfuscate_id(gateway_id))
+            return self.async_abort(reason="already_configured")
+
```

Comment:
In the future, update the config entry with the new ip address

Reply:
I am not sure if I understand what you mean. We use the gateway id to identify if an installation (physical hub) has been added already. Previously we relied on the username (email) for that, however some users have multiple hubs with the same emailaddress. 

And this way we can identify if an user has already configured a hub and not prompt them via DHCP discovery again.

Reply:
Never-mind, I was thinking you were connecting locally, but this isn't the case.